sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AspNetMvcHelper.cs

```diff
@@ -44,6 +49,8 @@ public static class AspNetMvcHelper
         /// </summary>
         public static bool IsControllerMethod(this IMethodSymbol methodSymbol) =>
             methodSymbol.GetEffectiveAccessibility() == Accessibility.Public &&
+            methodSymbol.MethodKind == MethodKind.Ordinary &&
```

Comment:
I would put this line before the effective accessibility check as it is simpler (faster).